### PR TITLE
Add tests/utils/scripts for running integration tests

### LINKED ISSUE #419: Create a framework for testing integration with Kubernetes
The [Proxy Privacy Project](https://github.com/runconduit/conduit/projects/1) is planning to rely extensively on Kubernetes infrastructure for its basic functionality.

In order to properly test it, we need to have a test framework that allows us to set up a Kubernetes cluster, install Conduit into it, verify the installation worked, make other changes to the cluster (e.g. create namespaces, create pods in those namespaces with and without the conduit proxy injected into them), test the behavior of the cluster in various scenerios (e.g. test that pods with the conduit proxy installed into them pick up changes to their TLS certificate when the CA certificate changes), and tear down the cluster. The important thing is that the tests are fully scriptable.

Initially it would be fine for this framework to only run in the latest version of minikube, but eventually it will be important to run the same tests in other platforms like GKE, AKS, EKS, as well as maybe older versions of minikube.

It would be nice of we could eventually run these tests to validate pull requests and/or the state of the master branch in CI.

### PR DESCRIPTION
**Pull request by klingerf**
Add a suite of integration tests in the `test/` directory, as well as utilities for testing in the `testutil/` directory.

You can use the `bin/test-run` script to run the full suite of tests, and the `bin/test-cleanup` script to cleanup after the tests.

The [TEST.md](https://github.com/runconduit/conduit/blob/kl/tests/TEST.md) file has more information about running tests.

@pcalcado, @franziskagoltz, and @rmars also contributed to this change. 🎉 

Fixes #419.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -146,6 +146,10 @@ bin/conduit stat deployments
 bin/conduit tap deploy emojivoto/voting
 ```
 
+### Testing
+
+For comprehensive testing, see the [test README](test/README.md).
```
Comment (siggy) [inline BUILD.md]: tioli: consider just in-lining `test/README.md` into `BUILD.md`

File: bin/test-cleanup [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+conduit_namespace=$1
```
Comment (siggy) [inline bin/test-cleanup]: confirming that `test-run` defaults to `conduit` namespace but `test-cleanup` intentionally does not?

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+```bash
+$ DOCKER_TRACE=1 bin/mkube bin/docker-build
+```
+
+That command also outputs the corresponding conduit binaries into the
+`target/cli` directory, and you can use the `bin/conduit` script to load those
+binaries when running tests. From the root of the conduit repo, run:
+
+```bash
+$ bin/test-run `pwd`/bin/conduit
```
Comment (siggy) [inline test/README.md]: related to my comment at the top, perhaps some mention of getting docker images into the environment that is to be tested? as written i think these instructions will only work in minikube.

File: bin/test-run [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+conduit_path=$1
+
+if [ -z "$conduit_path" ]; then
+    echo "usage: $(basename "$0") /path/to/conduit [namespace]" >&2
+    exit 64
+fi
+
+check_conduit_binary
+check_if_k8s_reachable
```
Comment (siggy) [inline bin/test-run]: related to other comments, any way to check for presence of correct docker images?

Comment (pcalcado) [reply bin/test-run]: I that this might be less relevant now that bb is a dockerhub image, it will be pulled by k8s if absent anyway

**Comment by pcalcado** [comment]
(btw I'm generally 👍  this but wasn't able to run it on my box just yet so didnt want to click approve)

File: test/README.md (hunk 3 above)
Comment (klingerf) [reply test/README.md]: There's a bullet point about this in the Prerequisites section of this doc:

```
- The conduit images you're trying to test have been built and are accessible
  to the Kubernetes cluster to which you are deploying
```

I can call that out more explicitly though, and will add a note about supported environments.

File: TEST.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+functionality via a series of end-to-end tests.
+
+## Prerequisites
+
+The integration test suite operates on your currently configured Kubernetes
+cluster. Prior to running the test suite, verify that:
+
+- The Conduit docker images you're trying to test have been built and are
+  accessible to the Kubernetes cluster to which you are deploying
+- The kubectl CLI has been configured to talk to that Kubernetes cluster
```
Comment (siggy) [inline TEST.md]: consider highlighting `kubectl`

File: TEST.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+- The Kubernetes cluster automatically provisions external IPs for external load
```
Comment (siggy) [inline TEST.md]: consider mentioning that GKE does this.

File: TEST.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  - [Rust](#rust)
+  - [Go](#go)
+  - [React](#react)
+
+# Integration tests
+
+The `test/` directory contains a test suite that can be run to validate Conduit
+functionality via a series of end-to-end tests.
+
+## Prerequisites
```
Comment (siggy) [inline TEST.md]: consider just calling out that this works in minikube and gke, and doesn't work in docker for mac + k8s.

File: TEST.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+The run the tests with:
+
+```bash
+cd proxy-init/integration_test
+eval $(minikube docker-env)
+./run_tests.sh
+```
+
+# Unit tests
```
Comment (siggy) [inline TEST.md]: i may be biased as i'm so used to running these unit tests, but consider putting them first in the doc. i think as a new user to the repo (and perhaps making a specific change to one part of the codebase) the first thing i like to do is to run these types of tests to get a baseline that that area is building/working.

**Comment by klingerf** [comment]
@siggy Thanks for the additional suggestions -- have updated in the latest commit. Re: docker for mac + k8s -- k8s support is only available in edge versions of docker for mac. I don't think we need to explicitly call out that we don't support an experimental environment.

File: TEST.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+  accessible to the Kubernetes cluster to which you are deploying
+- The `kubectl` CLI has been configured to talk to that Kubernetes cluster
+- The Kubernetes cluster automatically provisions external IPs for external load
+  balanced services (e.g. GKE), or the cluster is running in minikube
+- The `conduit` namespace that you're testing does not already exist
+- The repo's Go dependencies have been downloaded by running `bin/dep ensure`
+
+## Running tests
+
+You can use the `bin/test-run` script to run the full suite of tests.
```
Comment (rmars) [inline TEST.md]: TIOLI: should we add instructions on how to run a single test?

**Pull request merged**